pkg/kubernetes/pullsecrets.go

```diff
@@ -0,0 +1,78 @@
+package kubernetes
+
+import (
+	"context"
+
+	perrors "github.com/pkg/errors"
+)
+
+func (k *kubernetesDriver) EnsurePullSecret(
+	ctx context.Context,
+	pullSecretMetaData *PullSecretMetaData,
+	dockerConfigPath string,
+) error {
+	//todo: check if pull secret already exists, don't create it again
+	err := k.createPullSecret(ctx, *pullSecretMetaData, dockerConfigPath)
+	if err != nil {
+		return err
+	}
+
+	k.Log.Debugf("Pull secret '%s' created", pullSecretMetaData.pullSecretName)
+
+	err = k.addImagePullSecretsToServiceAccount(ctx, *pullSecretMetaData)
+	if err != nil {
+		return err
+	}
+	k.Log.Debugf("Pull secret '%s' added to service account '%s'",
+		pullSecretMetaData.pullSecretName, pullSecretMetaData.serviceAccount)
+
+	return nil
+}
+
+type PullSecretMetaData struct {
+	pullSecretName string
+	serviceAccount string
+}
+
+func (k *kubernetesDriver) createPullSecret(
+	ctx context.Context,
+	psMetaData PullSecretMetaData,
+	dockerConfigPath string,
+) error {
+	args := []string{
+		"create",
+		"secret",
+		"generic",
+		psMetaData.pullSecretName,
+		"--from-file=.dockerconfigjson=" + dockerConfigPath,
```

Comment:
that is not correct unfortunately, you need to extract a token (like devspace / devpod does) and then create a config out of that